Mathlib/LinearAlgebra/Matrix/Trace.lean

```diff
@@ -196,6 +199,11 @@ lemma trace_submatrix_succ {n : ℕ} [NonUnitalNonAssocSemiring R]
   rw [← (finSuccEquiv n).symm.sum_comp]
   simp
 
+theorem trace_permutation [DecidableEq n] [AddCommMonoidWithOne R] (σ : Equiv.Perm n) :
+    Matrix.trace (σ.toPEquiv.toMatrix : Matrix n n R) = (Function.fixedPoints σ).ncard := by
```

Comment:
If we are going to start a new file, may I suggest an
```lean
abbrev Equiv.Perm.permMatrix (σ : Equiv.Perm n) : Matrix n n R := σ.toPEquiv.toMatrix
```
And then the existing
```lean
-- Mathlib/LinearAlgebra/Matrix/Determinant.lean
275:    Matrix.det (σ.toPEquiv.toMatrix : Matrix n n R) = Perm.sign σ := by
276:  rw [← Matrix.mul_one (σ.toPEquiv.toMatrix : Matrix n n R), PEquiv.toPEquiv_mul_matrix,

-- Mathlib/Data/Matrix/PEquiv.lean
97:    (M : Matrix m n α) : f.toPEquiv.toMatrix * M = M.submatrix f id := by
102:theorem mul_toPEquiv_toMatrix {m n α : Type*} [Fintype n] [DecidableEq n] [Semiring α] (f : n ≃ n)
103:    (M : Matrix m n α) : M * f.toPEquiv.toMatrix = M.submatrix id f.symm :=
107:#align pequiv.mul_to_pequiv_to_matrix PEquiv.mul_toPEquiv_toMatrix
143:    (Equiv.swap i j).toPEquiv.toMatrix =
174:theorem equiv_toPEquiv_toMatrix [DecidableEq n] [Zero α] [One α] (σ : Equiv n n) (i j : n) :
175:    σ.toPEquiv.toMatrix i j = (1 : Matrix n n α) (σ i) j :=
177:#align pequiv.equiv_to_pequiv_to_matrix PEquiv.equiv_toPEquiv_toMatrix
```
could also move to this new file.

Reply:
`mul_toPEquiv_toMatrix` and `equiv_toPEquiv_toMatrix` are about `Equiv.toPEquiv`, not `Equiv.Perm.toEquiv`. Moved `det`, crediting Anne for https://github.com/leanprover-community/mathlib/pull/1816